pyEDAA/UCIS/Cobertura.py

```diff
@@ -136,56 +152,80 @@ class Coverage:
+		self.sources.add(source)
+
+	def addPackage(self, package: Package) -> None:
+		assert package.name not in self.packages, "Duplicated package name"
+		self.packages[package.name] = package
 
-	def add_statement(self, source: str, file: str, line: int, count: int) -> None:
+	def addStatement(self, source: str, file: str, line: int, count: int) -> None:
 		try:
-			self.packages[source].add_statement(file, file, line, count)
+			self.packages[source].addStatement(file, file, line, count)
 		except KeyError:
 			self.packages[source] = Package(file)
-			self.packages.get(source).add_statement(file, file, line, count)
+			self.packages.get(source).addStatement(file, file, line, count)
 
 		self.sources.add(source)
 
-		self.lines_valid += 1
+		self.linesValid += 1
 
 		if count:
-			self.lines_covered += 1
+			self.linesCovered += 1
 
-	def add_branch(self) -> None:
-		pass
+	def refreshStatistics(self) -> None:
+		self.linesValid = 0
+		self.linesCovered = 0
 
-	def get_xml(self) -> etree.Element:
-		coverage_node = etree.Element("coverage")
-		coverage_node.attrib["version"] = "5.5"
-		coverage_node.attrib["timestamp"] = str(int(time()))
-		coverage_node.attrib["branches-valid"] = "0"
-		coverage_node.attrib["branches-covered"] = "0"
-		coverage_node.attrib["branch-rate"] = "0"
-		coverage_node.attrib["complexity"] = "0"
-		coverage_node.attrib["lines-valid"] = str(self.lines_valid)
-		coverage_node.attrib["lines-covered"] = str(self.lines_covered)
-		coverage_node.attrib["line-rate"] = str(self.lines_covered / self.lines_valid)
+		for package in self.packages.values():
+			package.refreshStatistics()
+			self.linesCovered += package.linesCovered
+			self.linesValid += package.linesValid
 
-		sources_node = etree.Element("sources")
+	def getXml(self) -> bytes:
+		self.refreshStatistics()
+
+		coverageNode = etree.Element("coverage")
+		coverageNode.attrib["version"] = "5.5"
+		coverageNode.attrib["timestamp"] = str(int(time()))
+		coverageNode.attrib["branches-valid"] = "0"
+		coverageNode.attrib["branches-covered"] = "0"
+		coverageNode.attrib["branch-rate"] = "0"
+		coverageNode.attrib["complexity"] = "0"
+
+		sourcesNode = etree.Element("sources")
 
 		for source in self.sources:
-			etree.SubElement(sources_node, "source").text = source
+			etree.SubElement(sourcesNode, "source").text = source
+
+		coverageNode.append(sourcesNode)
 
-		coverage_node.append(sources_node)
+		packagesNode = etree.Element("packages")
 
-		packages_node = etree.Element("packages")
+		for package in self.packages.values():
+			packagesNode.append(package.getXmlNode())
 
-		for package_name in self.packages:
-			packages_node.append(self.packages[package_name].get_xml_node())
+		coverageNode.append(packagesNode)
 
-		coverage_node.append(packages_node)
+		coverageNode.attrib["lines-valid"] = str(self.linesValid)
+		coverageNode.attrib["lines-covered"] = str(self.linesCovered)
+
+		try:
+			coverageNode.attrib["line-rate"] = str(
```

Comment:
Like a previous code part.

Reply:
Fixed in https://github.com/edaa-org/pyEDAA.UCIS/pull/7/commits/a3989380939cc40dad5c58f093cfe2d9e3aa499f

Reply:
Checked.